internal/controllers/bundledeployment/bundledeployment.go

```diff
@@ -145,21 +186,31 @@ func (c *controller) validateConfig() error {
 type controller struct {
 	cl client.Client
 
-	handler          Handler
-	provisionerID    string
-	acg              helmclient.ActionClientGetter
-	storage          storage.Storage
-	releaseNamespace string
+	handler                   Handler
+	bundleDeploymentProcessor Processor
+	provisionerID             string
+	acg                       helmclient.ActionClientGetter
+	storage                   storage.Storage
+	releaseNamespace          string
 
+	unpacker          unpackersource.Unpacker
 	controller        crcontroller.Controller
+	finalizers        crfinalizer.Finalizers
 	dynamicWatchMutex sync.RWMutex
 	dynamicWatchGVKs  map[schema.GroupVersionKind]struct{}
 }
 
-//+kubebuilder:rbac:groups=core.rukpak.io,resources=bundledeployments,verbs=list;watch
-//+kubebuilder:rbac:groups=core.rukpak.io,resources=bundledeployments/status,verbs=update;patch
+// TODO: recheck rbac
 //+kubebuilder:rbac:groups=core.rukpak.io,resources=bundledeployments/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core.rukpak.io,resources=bundledeployments,verbs=list;watch;update;patch
```

Comment:
Are the `update` and `patch` verbs actually necessary here? I see that we had them for `Bundles`, so maybe so? Only thing I can think of is that this is necessary to update finalizers and somehow the `bundledeployments/finalizers` permission is insufficient or not doing what we think it does?

Reply:
Had added these verbs because they were in Bundles. Will test this again and get back.

Reply:
Looks like they are not required. Fixed.